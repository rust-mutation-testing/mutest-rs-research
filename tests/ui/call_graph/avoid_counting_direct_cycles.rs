//@ print-call-graph
//@ print-targets
//@ stdout
//@ stderr: empty

fn recursive_function(i: usize) {
    if i == 1 { return; }
    recursive_function(i - 1);
}

fn other_recursive_function(i: usize) {
    recursive_function(i);
}

#[test]
fn test() {
    recursive_function(3);
    other_recursive_function(3);
}
