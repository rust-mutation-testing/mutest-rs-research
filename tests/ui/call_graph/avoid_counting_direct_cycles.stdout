
@@@ call graph @@@

entry points:

test test
  -> recursive_function at tests/ui/call_graph/avoid_counting_direct_cycles.rs:6:1: 6:32 (#0)
  -> other_recursive_function at tests/ui/call_graph/avoid_counting_direct_cycles.rs:11:1: 11:38 (#0)

nested calls at distance 1:

recursive_function at tests/ui/call_graph/avoid_counting_direct_cycles.rs:6:1: 6:32 (#0)
  -> recursive_function at tests/ui/call_graph/avoid_counting_direct_cycles.rs:6:1: 6:32 (#0)
other_recursive_function at tests/ui/call_graph/avoid_counting_direct_cycles.rs:11:1: 11:38 (#0)
  -> recursive_function at tests/ui/call_graph/avoid_counting_direct_cycles.rs:6:1: 6:32 (#0)

@@@ targets @@@

tests -(0)-> recursive_function at tests/ui/call_graph/avoid_counting_direct_cycles.rs:6:1: 6:32 (#0)
  (0) test

tests -(0)-> other_recursive_function at tests/ui/call_graph/avoid_counting_direct_cycles.rs:11:1: 11:38 (#0)
  (0) test

targets: 2 total; 2 safe; 0 unsafe (0 tainted)
